sefaria/sheets.py

```diff
@@ -414,7 +423,19 @@ def save_sheet(sheet, user_id, search_override=False, rebuild_nodes=False):
 
 	if rebuild_nodes:
 		sheet = rebuild_sheet_nodes(sheet)
-	db.sheets.update({"id": sheet["id"]}, sheet, True, False)
+
+	if new_sheet:
+		# mongo enforces a unique sheet id, get a new id until a unique one has been found
+		while True:
+			try:
+				sheet["id"] = next_sheet_id()
+				db.sheets.insert_one(sheet)
+				break
+			except DuplicateKeyError:
+				pass
+
+	else:
+		db.sheets.update({"id": sheet["id"]}, sheet, False, False)
```

Comment:
`update` is deprecated.  
```suggestion
		db.sheets.update_one({"id": sheet["id"]}, sheet)
```